server-side/controllers/user.controller.js

```diff
@@ -15,7 +15,8 @@ export const getUsers = async (req, res,next) => {
 };
 
 export const getUserById = async (req, res,next) => {
-  const id = req.params.id;
+  let id = req.params?.id;
```

Comment:
if id is optional,please check if it has value

Reply:
This code is not related to me,
But the girl who wrote it did not agree to fix it and she claimed that it works

Reply:
I made a combined version of the access option both by http requests and by internal use of the server

Reply:
please use Destructuring Assignment:
 const {id} = req.params;

Reply:
I have already made changes because the function should be suitable for both API call and server internal use, so the line of code looks like this   const id = req.params;

Reply:
function in controller shouldn't be used for internal server use.
controller is for Rest API. functions that you want to use both you can write in service.
but if you said you can't change it so I'll approve it